### PR TITLE
rust 1.65.0

### PR DESCRIPTION
**Pull request by darthShadow**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by Bo98** [comment]
Cargo needs updating to 0.66.0, though there isn't a git tag yet.

**Comment by darthShadow** [comment]
Updated it with the non-existent tag (but the crate is released: https://crates.io/crates/cargo/0.66.0) and the corresponding commit.

**Comment by chenrui333** [comment]
🏃

**Comment by Porkepix** [comment]
I don't know what's the meaning of these tags, but they would rather seems ahead.
Wouldn't that be more accurate to use the branches that have the same name as the corresponding rust release? These branches are frozen at the same time, and that's also what the changelog refers to, see:
- https://github.com/rust-lang/cargo/tree/rust-1.65.0 (its last commit is the one pointed to in link in the formula's comment)
- https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md : the diff under every release points to the branch with a name corresponding with the release of rust of the same version;

I don't know what these 0.x tags are here for, means, or meant, but they doesn't looks like it should be use here, in regards of the Rust version?

I'll try to drop by the IRC channel asking if someone knows about it, though that's an unofficial one and I don't use Discord which seems to be the one highlighted on the site.

**Comment by darthShadow** [comment]
Linked issue for anyone who finds this later (thanks to @Bo98): https://github.com/rust-lang/cargo/issues/11345

**Comment by Bo98** [comment]
> Wouldn't that be more accurate to use the branches that have the same name as the corresponding rust release? These branches are frozen at the same time, and that's also what the changelog refers to, see:

No, branches are not fixed targets. They can be updated for .1 patches releases.

File: Formula/rust.rb [hunk 1]
```diff
@@ -4,14 +4,14 @@ class Rust < Formula
...
-    sha256 "b3cd9f481e1a2901bf6f3808d30c69cc4ea80d93c4cc4e2ed52258b180381205"
+    url "https://static.rust-lang.org/dist/rustc-1.65.0-src.tar.gz"
+    sha256 "5828bb67f677eabf8c384020582b0ce7af884e1c84389484f7f8d00dd82c0038"
 
     # From https://github.com/rust-lang/rust/tree/#{version}/src/tools
     resource "cargo" do
       url "https://github.com/rust-lang/cargo.git",
-          tag:      "0.65.0",
-          revision: "387270bc7f446d17869c7f208207c73231d6a252"
+          tag:      "0.66.0",
```
Comment (Bo98) [inline Formula/rust.rb:13]:
```suggestion
          # tag:      "0.66.0",
```

Can do this for now. It'll get it based on the commit.

Comment (darthShadow) [reply Formula/rust.rb]: Done, thanks.

**Comment by Porkepix** [comment]
> > I don't know what these 0.x tags are here for, means, or meant, but they doesn't looks like it should be use here, in regards of the Rust version?
> 
> Rust 1.X.0 maps to Cargo 0.(X + 1).0. Patch releases aren't always in sync depending on the nature of the need for a patch (Cargo is also used as a library) but ultimately remain compatible with each other regardless.

That's what I was told on `##rust` on IRC. But they seemed to agree the branch made more sense than the tag.
Also, doesn't it make it redundant to use both the tag and the commit/revision? The comment in the formula point to a place only referencing a commit, so it could also be the commit alone that would be kept?

**Comment by Porkepix** [comment]
> > make it redundant to use both the tag and the commit/revision?
> 
> No, because otherwise a malicious actor could retag and ship malware through Homebrew.

That's why I was rather thinking about the commit/revision only, because that's what the rust repo points to.

**Comment by SMillerDev** [comment]
> That's why I was rather thinking about the commit/revision only, because that's what the rust repo points to.

There is a mechanism to mark a specific commit/revision as an important point in development worth referencing to though. It's git tags.

**Comment by darthShadow** [comment]
The tag is now available but I am not sure whether I should uncomment it now or wait for the CI to finish since it's been running for a day now.

**Comment by branchvincent** [comment]
Let's wait for CI to finish

**Comment by darthShadow** [comment]
Looks like the CI finally finished (with a failure) so I have uncommented the tag now.

**Comment by darthShadow** [comment]
Bump, not stale.

**Comment by chenrui333** [comment]
> Looks like the CI finally finished (with a failure), so I have uncommented the tag now.

What are those failures? I could not see the action run now (somehow)

**Comment by darthShadow** [comment]
I think it failed because the CI-long-timeout label was removed before it could finish. I don't recall anything specific otherwise in the failure log.

**Comment by SMillerDev** [comment]
> I think it failed because the CI-long-timeout label was removed before it could finish. I don't recall anything specific otherwise in the failure log.

That label only changes settings at the start of the build. There must have been other failures.

**Comment by darthShadow** [comment]
> There must have been other failures.

Ah, apologies, I must have missed them then. I wasn't aware the previous action runs wouldn't be available anymore (https://github.com/Homebrew/homebrew-core/actions/runs/3451877310 returns a 404 now), otherwise I wouldn't have updated the PR.

**Comment by chenrui333** [comment]
@darthShadow looks your PR does not allow forced-push update, gonna create a new PR for the new run, thanks!

**Comment by chenrui333** [comment]
#116480

**Pull request closed without merging**